**Windows** [Build 4583](https://ci.wildfly.org/viewLog.html?buildId=15946&buildTypeId=WF_PullRequest_Windows) outcome was **FAILURE** using a merge of bd5615fdda23220f89b5fb50d26cb480676a9998
Summary: Tests failed: 2 (2 new), passed: 3313, ignored: 195 Build time: 01:17:53
### Failed tests
```
org.jboss.as.security.VaultToolTestCase.testVaultTool: java.lang.AssertionError: Exit status is equal to 0 expected:<0> but was:<1>
	at org.jboss.as.security.VaultToolTestCase.doTestVaultTool(VaultToolTestCase.java:112)
	at org.jboss.as.security.VaultToolTestCase.testVaultTool(VaultToolTestCase.java:85)
------- Stderr: -------
Problem occurred:
java.nio.file.InvalidPathException: Illegal char <:> at index 2: /C:/BuildAgent/work/a31d203e70e89f90/security/subsystem/target/test-classes/org/jboss/as/security/vault.keystore
	at sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at sun.nio.fs.WindowsPath.parse(WindowsPath.java:94)
	at sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:255)
	at java.nio.file.Paths.get(Paths.java:84)
	at org.jboss.as.security.vault.VaultSession.getVaultOptionsMap(VaultSession.java:225)
	at org.jboss.as.security.vault.VaultSession.initSecurityVault(VaultSession.java:193)
	at org.jboss.as.security.vault.VaultSession.startVaultSession(VaultSession.java:214)
	at org.jboss.as.security.vault.VaultTool.execute(VaultTool.java:193)
	at org.jboss.as.security.vault.VaultTool.main(VaultTool.java:83)
	at org.jboss.as.security.VaultToolTestCase.doTestVaultTool(VaultToolTestCase.java:110)
	at org.jboss.as.security.VaultToolTestCase.testVaultTool(VaultToolTestCase.java:85)
	at sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at sun.nio.fs.WindowsPath.parse(WindowsPath.java:94)
	at sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:255)
	at java.nio.file.Paths.get(Paths.java:84)
	at org.jboss.as.security.VaultToolTestCase.tearDown(VaultToolTestCase.java:210)


org.jboss.as.security.VaultToolTestCase.testNoKeyStoreFile: java.lang.AssertionError: Exit status is equal to 0 expected:<0> but was:<1>
	at org.jboss.as.security.VaultToolTestCase.doTestVaultTool(VaultToolTestCase.java:112)
	at org.jboss.as.security.VaultToolTestCase.testNoKeyStoreFile(VaultToolTestCase.java:96)
------- Stderr: -------
Problem occurred:
java.nio.file.InvalidPathException: Illegal char <:> at index 2: /C:/BuildAgent/work/a31d203e70e89f90/security/subsystem/target/test-classes/org/jboss/as/security/vault.keystore
	at sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at sun.nio.fs.WindowsPath.parse(WindowsPath.java:94)
	at sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:255)
	at java.nio.file.Paths.get(Paths.java:84)
	at org.jboss.as.security.vault.VaultSession.getVaultOptionsMap(VaultSession.java:225)
	at org.jboss.as.security.vault.VaultSession.initSecurityVault(VaultSession.java:193)
	at org.jboss.as.security.vault.VaultSession.startVaultSession(VaultSession.java:214)
	at org.jboss.as.security.vault.VaultTool.execute(VaultTool.java:193)
	at org.jboss.as.security.vault.VaultTool.main(VaultTool.java:83)
	at org.jboss.as.security.VaultToolTestCase.doTestVaultTool(VaultToolTestCase.java:110)
	at org.jboss.as.security.VaultToolTestCase.testNoKeyStoreFile(VaultToolTestCase.java:96)
	at sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at sun.nio.fs.WindowsPath.parse(WindowsPath.java:94)
	at sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:255)
	at java.nio.file.Paths.get(Paths.java:84)
	at org.jboss.as.security.VaultToolTestCase.tearDown(VaultToolTestCase.java:210)


```

--- a/security/subsystem/src/main/java/org/jboss/as/security/vault/VaultInteractiveSession.java
+++ b/security/subsystem/src/main/java/org/jboss/as/security/vault/VaultInteractiveSession.java
@@ -68,7 +68,7 @@ public class VaultInteractiveSession {
 
             String ic = console.readLine(SecurityLogger.ROOT_LOGGER.enterIterationCount() + " ");
             iterationCount = Integer.parseInt(ic);
-            vaultNISession = new VaultSession(keystoreURL, new String(keystorePasswd), encDir, salt, iterationCount);
+            vaultNISession = new VaultSession(keystoreURL, new String(keystorePasswd), encDir, salt, iterationCount, true);
 
             while (keystoreAlias == null || keystoreAlias.length() == 0) {
                 keystoreAlias = console.readLine(SecurityLogger.ROOT_LOGGER.enterKeyStoreAlias() + " ");

--- a/security/subsystem/src/main/java/org/jboss/as/security/vault/VaultSession.java
+++ b/security/subsystem/src/main/java/org/jboss/as/security/vault/VaultSession.java
@@ -115,7 +115,9 @@ public final class VaultSession {
 
         File f = new File(keystoreURL);
         if (!f.exists()) {
-            throw SecurityLogger.ROOT_LOGGER.keyStoreDoesnotExistWithExample(keystoreURL, keystoreURL);
+            if (!createKeystore) {
+                throw SecurityLogger.ROOT_LOGGER.keyStoreDoesnotExistWithExample(keystoreURL, keystoreURL);
+            }
         } else if (!f.canWrite() || !f.isFile()) {
             throw SecurityLogger.ROOT_LOGGER.keyStoreNotWritable(keystoreURL);
         }
@@ -218,7 +220,7 @@ public final class VaultSession {
         options.put(PicketBoxSecurityVault.SALT, salt);
         options.put(PicketBoxSecurityVault.ITERATION_COUNT, Integer.toString(iterationCount));
         options.put(PicketBoxSecurityVault.ENC_FILE_DIR, encryptionDirectory);
-        if (createKeystore) {
+        if (createKeystore && !new File(keystoreURL).exists()) {
             options.put(PicketBoxSecurityVault.CREATE_KEYSTORE, Boolean.toString(createKeystore));
         }
         return options;

--- a/security/subsystem/src/test/java/org/jboss/as/security/VaultToolTestCase.java
+++ b/security/subsystem/src/test/java/org/jboss/as/security/VaultToolTestCase.java
@@ -80,18 +80,25 @@ public class VaultToolTestCase {
 
   @Test
   public void testVaultTool() throws Exception {
-    doTestVaultTool(false);
+    doTestVaultTool(true, false);
   }
 
   @Test
   @Ignore
   public void testVaultFallback() throws Exception {
-    doTestVaultTool(true);
+    doTestVaultTool(true, true);
   }
 
-  private void doTestVaultTool(boolean testFallbackFromIncorrectPasswordValue) throws Exception {
+  @Test
+  public void testNoKeyStoreFile() throws Exception {
+    doTestVaultTool(false, false);
+  }
 
-    createAndFillKeystore(KEYSTORE_URL_VALUE);
+  private void doTestVaultTool(boolean prepareKeystore, boolean testFallbackFromIncorrectPasswordValue) throws Exception {
+
+    if (prepareKeystore) {
+        createAndFillKeystore(KEYSTORE_URL_VALUE);
+    }
 
     // Replaces standard output with a ByteArrayOutputStream to parse the output later.
     System.setOut(new PrintStream(SYSTEM_OUT));
@@ -156,9 +163,7 @@ public class VaultToolTestCase {
     addAll(args, "-b", BLOCK_NAME);
     addAll(args, "-a", ATTRIBUTE_NAME);
     addAll(args, "-x", VALUE_TO_STORE);
-    /* removed as temporary workaround until PB 4.9.0.Beta2 is pulled to WF-CORE
     addAll(args, "-t");
-    */
     return args.toArray(new String[0]);
   }
 
@@ -200,6 +205,10 @@ public class VaultToolTestCase {
   public void tearDown() throws Exception {
     System.setSecurityManager(null); // or save and restore original
     cleanDirectory(ENC_FILE_DIR_VALUE);
+    File keyStoreFile = new File(KEYSTORE_URL_VALUE);
+    if (keyStoreFile.exists()) {
+        keyStoreFile.delete();
+    }
   }
 
   /**

